app/models/lookup_keys/variable_lookup_key.rb

```diff
@@ -16,8 +15,19 @@ def param_class
     puppetclass
   end
 
-  def self.humanize_class_name
-    "Smart variable"
+  def self.humanize_class_name(options = nil)
+    if options.present?
+      super
+    else
+      "Smart variable"
+    end
+  end
+
+  def validate_puppet_class_presence
+    # Variable lookup key can be created through puppet class.
+    # If this is a new puppet class the id is still nil so validating presence on puppetclass_id will fail.
+    # Adding the validation on the puppet class itself will not show up in the UI because the displayed field is puppetclass_id.
+    errors[:puppetclass_id] = _("can't be blank") unless self.puppetclass.present?
```

Comment:
I changed the validation because this test was failing: https://github.com/theforeman/foreman/blob/develop/test/models/puppetclass_test.rb#L46
I don't think we can actually get to this case through the api/ui but the test makes me think there may be some way to so this through the importer so I changed the validation.
If this really isn't possible I'd prefer to change the test and move the validation back to:
`validates :puppetclass_id, :presence => true`